src/qibolab/platform.py

```diff
@@ -178,9 +216,30 @@ def execute_pulse_sequences(self, sequences: List[PulseSequence], options: Execu
             **kwargs: May need them for something
         Returns:
             Readout results acquired by after execution.
-
         """
-        return self._execute("play_sequences", sequences, options, **kwargs)
+        options = self.settings.apply(options)
+
+        duration = sum(seq.duration for seq in sequences)
+        time = (duration + len(sequences) * options.relaxation_time) * options.nshots * 1e-9
+        log.info(f"Minimal execution time (unrolling): {time}")
+
+        try:
+            return self._execute("play_sequences", sequences, options, **kwargs)
```

Comment:
Maybe you could avoid the `getattr` making a simple `enum`, and switch from one to the other.

The function will be something like:
```py
class PlayKind(Enum):
    ONE = auto()
    SEQUENCE = auto()

def play_(instrument, kind):
    if kind is PlayKind.ONE:
        return instrument.play
    return instrument.play_sequence
```
and use in place of the `getattr` and strings.

Reply:
If the issue is the `getattr`, isn't it simpler to just have a bool `use_play_sequences` and do:
```py
if use_play_sequences:
    instrument.play_sequences(...)
else:
    instrument.play(...)
```
Essentially replacing the `Enum` with `bool` as there are only two options.

Also, note that this only exists because of `instrument.play_sequences`. However in practice this method is not implemented for any instrument. It is kind of replaced by the unrolling implemented here, however I still keep it in case we discover that some instrument really benefits from having a specialized unrolling method.

Reply:
> Essentially replacing the `Enum` with `bool` as there are only two options.

Yes, with two options that's always possible. And in this case you do not lose much from the names (`ONE` was not even that meaningful).

Just use `sequences=False` (instead of the whole `use_play_sequences`, since `play_` is a bit redundant, and `use_` is slightly more specific than `flag_` or `is_`, and it's implicit in the `bool` type).

> Also, note that this only exists because of `instrument.play_sequences`. However in practice this method is not implemented for any instrument. It is kind of replaced by the unrolling implemented here, however I still keep it in case we discover that some instrument really benefits from having a specialized unrolling method.

I have the feeling that we'll never need that. Especially after the discussions of the other week on the individual drivers.
In any case, an optional flag is even easier to understand and drop than anything else (the current opaque `getattr`, and the verbose `Enum`).